src/lib/stitching/gravity/stitching.ts

```diff
@@ -81,29 +117,79 @@ export const gravityStitchingEnvironment = (
             })
           },
         },
-        formattedEndAt: {
+        distanceToOpen: {
+          fragment: gql`
+            ... on ViewingRoom {
+              startAt
+            }
+		  `,
+          resolve: ({ startAt: _startAt }, { short = false }) => {
+            if (_startAt === null) {
+              return null
+            }
+
+            const startAt = moment(_startAt)
+            const now = moment()
+
+            if (startAt < now) {
+              return null
+            }
+
+            if (short === false && startAt > now.clone().add(30, "days")) {
+              return null
+            }
+
+            const distance = moment.duration(startAt.diff(now))
+            return distance
+              .locale(
+                short
+                  ? LocaleEnViewingroomRelativeShort
+                  : LocaleEnViewingroomRelativeLong
+              )
+              .humanize(short, { ss: 1, d: 31 })
+          },
+        },
+        distanceToClose: {
           fragment: gql`
             ... on ViewingRoom {
               startAt
               endAt
             }
           `,
-          resolve: ({ startAt: _startAt, endAt: _endAt }) => {
+          resolve: (
+            { startAt: _startAt, endAt: _endAt },
+            { short = false }
+          ) => {
+            if (_startAt === null || _endAt === null) {
```

Comment:
Maybe something like this could simplify this conditional?

```suggestion
            if (_startAt && _endAt) {
```